modules/contact/test/tests/bouncing-block-contact/frictionless-weighted-gap-mixed-basis.i

```diff
@@ -0,0 +1,173 @@
+starting_point = 2e-1
+offset = 1e-2
+
+[GlobalParams]
+  displacements = 'disp_x disp_y'
+  diffusivity = 1e0
+  scaling = 1e0
+[]
+
+[Mesh]
+  file = long-bottom-block-1elem-blocks.e
+  second_order = true
+  patch_update_strategy = always
+[]
+
+[Variables]
+  [./disp_x]
+    block = '1 2'
+    order = SECOND
+  [../]
+  [./disp_y]
+    block = '1 2'
+    order = SECOND
+  [../]
+  [./normal_lm]
+    block = 3
+  [../]
+[]
+
+[ICs]
+  [./disp_y]
+    block = 2
+    variable = disp_y
+    value = ${fparse starting_point + offset}
+    type = ConstantIC
+  [../]
+[]
+
+[Kernels]
+  [./disp_x]
+    type = MatDiffusion
+    variable = disp_x
+  [../]
+  [./disp_y]
+    type = MatDiffusion
+    variable = disp_y
+  [../]
+[]
+
+
+[Constraints]
+  [./weighted_gap_lm]
+    type = ComputeWeightedGapLMMechanicalContact
+    primary_boundary = 20
+    secondary_boundary = 10
+    primary_subdomain = 4
+    secondary_subdomain = 3
+    variable = normal_lm
+    secondary_variable = disp_x
+    secondary_disp_y = disp_y
+    use_displaced_mesh = true
+  [../]
+  [./ncp_lm]
+    type = ApplyPenetrationConstraintLMMechanicalContact
+    secondary = 10
+    primary = 20
```

Comment:
Should we make these names consistent?

```

    primary_boundary = 20
     secondary_boundary = 10
```

Reply:
So `ApplyPenetrationConstraintLMMechanicalContact` is actually a `NodeFaceConstraint` and it has has the `secondary` and `primary` params for many years. It is fine for them to do this because they don't also have lower-dimensional subdomains associated with their object like `MortarConstraints` do. So while the discrepancy is obnoxious I don't think we should change it at this point. I will note that our users of mortar contact (and contact in general) generally go through the `ContactAction` so this discrepancy is hidden from them